tests/OSPSuite.Core.Tests/Domain/ExpressionProfileBuildingBlockSpecs.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Linq;
+using OSPSuite.BDDHelper;
+using OSPSuite.BDDHelper.Extensions;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.Helpers;
+
+namespace OSPSuite.Core.Domain
+{
+   public class concern_for_ExpressionProfileBuildingBlock : ContextSpecification<ExpressionProfileBuildingBlock>
+   {
+      protected override void Context()
+      {
+         sut = new ExpressionProfileBuildingBlock();
+      }
+   }
+
+   public class when_updating_properties_of_building_block : concern_for_ExpressionProfileBuildingBlock
+   {
+      private ExpressionProfileBuildingBlock _expressionProfileBuildingBlock;
+      private ICloneManager _cloneManager;
+      private IObjectBaseFactory _objectBaseFactory;
+      private IDataRepositoryTask _repositoryTask;
+      private IDimensionFactory _dimensionFactory;
+
+      protected override void Context()
+      {
+         base.Context();
+         _dimensionFactory = new DimensionFactoryForIntegrationTests();
+         _dimensionFactory.AddDimension(DomainHelperForSpecs.NoDimension());
+         _objectBaseFactory = new ObjectBaseFactoryForSpecs(_dimensionFactory);
+         _repositoryTask = new DataRepositoryTask();
+         _expressionProfileBuildingBlock = new ExpressionProfileBuildingBlock();
+         _cloneManager = new CloneManagerForBuildingBlock(_objectBaseFactory, _repositoryTask);
+         _expressionProfileBuildingBlock.Name = "Molecule|Species|Name";
+         _expressionProfileBuildingBlock.Type = ExpressionTypes.MetabolizingEnzyme;
+         _expressionProfileBuildingBlock.Type = ExpressionTypes.MetabolizingEnzyme;
+         _expressionProfileBuildingBlock.PKSimVersion = 11;
+         _expressionProfileBuildingBlock.Add(new ExpressionParameter().WithName("name1"));
+      }
+
+      protected override void Because()
+      {
+         sut.UpdatePropertiesFrom(_expressionProfileBuildingBlock, _cloneManager);
+      }
+
+      [Observation]
+      public void the_updated_expression_profile_should_have_properties_set()
+      {
+         sut.Name.ShouldBeEqualTo("Molecule|Species|Name");
+         sut.Type.ShouldBeEqualTo(ExpressionTypes.MetabolizingEnzyme);
+         sut.PKSimVersion.ShouldBeEqualTo(11);
+         sut.Count().ShouldBeEqualTo(1);
+      }
+   }
+
+   //commenting out, as long as we do not have an icon based on type
+   /*
+   public class when_reading_the_icon_name_for_the_building_block : concern_for_ExpressionProfileBuildingBlock
```

Comment:
why is this still commented out? There is the icons defined in the type right?

Reply:
added back in